LogicalTypes.md

```diff
@@ -670,6 +681,8 @@ optional group array_of_arrays (LIST) {
 
 #### Backward-compatibility rules
```

Comment:
It is nice that we are including rules for how to interpret existing data, but can we also have guidelines for writers such that we converge on a single rule for future data? It would be nice to reduce the existing variability in the long term.

Reply:
The current spec (and it didn't change with this PR) specifies the "3-level" structure and the "unannotated" (aka. "1-level") list types as ones to be supported. So, according to the spec, a writer either use the first or the second one but cannot use both in the same schema.

Since only the "3-level" structure is capable of properly describing nullability at any level, this should be the preferred one.

We may start a discussion to deprecate the "1-level" list but I think this would be a different topic. This PR is about clarifying the backward compatibility rules only.

Reply:
Okay, so I've suggested an addition below to make this extra clear.